packages/expo-dev-launcher/bundle/components/UrlInput.tsx

```diff
@@ -9,11 +9,30 @@ import Button from './Button';
 type Props = {
   onPress: (url: string) => void;
 };
+
+const urlPattern = new RegExp(
+  '^((\\w+)?:\\/\\/)?' + // protocol
+  '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.?)+([a-z]{2,})?|' + // domain name
+  '((\\d{1,3}\\.){3}\\d{1,3}))' + // OR ip (v4) address
+  '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*' + // port and path
+  '(\\?[;&a-z\\d%_.~+=-]*)?' + // query string
+    '(\\#[-a-z\\d_]*)?$', // fragment locator
+  'i'
+);
```

Comment:
Have you found this pattern on the internet?
```suggestion
const urlPattern = new RegExp(
  '^((\\w+)?:\\/\\/)?' + // protocol
  '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.?)+([a-z]{2,})?|' + // domain name
  '((\\d{1,3}\\.){3}\\d{1,3}))' + // OR ip (v4) address
  '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*' + // port and path
  '(\\?[;&a-z\\d%_.~+=-]*)?' + // query string
  '(\\#[-a-z\\d_]*)?$', // fragment locator
  'i'
);
```

Reply:
Yep, https://stackoverflow.com/questions/5717093/check-if-a-javascript-string-is-a-url. However, I modified this slightly.